### DISCUSSION TITLE
Error decrypting message: Session key decryption failed. at yc.decryptSessionKeys

### DISCUSSION THREAD

**Discussion by plabelclothing** [openpgpjs/openpgpjs#1544] [Q&A]
Hi,

v.5.3.0

i can't decrypt the message(encrypted ZIP file). Unfortunately, i can't send a private key
In an older version(3.0.9) everything was working 

`    const publicKey = await openpgp.readKey({armoredKey: publicKeyEncode});
    const privateKey = await openpgp.readPrivateKey({armoredKey: privateKeyEncode});

    const message = await openpgp.readMessage({ binaryMessage });
    const decrypted = await openpgp.decrypt({ decryptionKeys: privateKey, verificationKeys: publicKey, message, format: 'binary' });
    console.log(decrypted.data); // Uint8Array
    console.log(await decrypted.signatures[0].verified); // Boolean`

encrypted file:
[https://drive.google.com/file/d/1zn-oEUf67BlzT0CCeeIerUSMkD2rmvk6/view?usp=sharing](url)

**Comment by twiss** [comment]
Hi, the shared zip file seems to be corrupted, so I can't open it, but most likely you're having a similar issue as https://github.com/openpgpjs/openpgpjs/discussions/1520, see the answer there for some suggestions.

**Reply by plabelclothing** [reply]
Hi, it isn't corrupted, but it's encrypted

**Reply by plabelclothing** [reply]
Our partners encrypt zip and send to SFTP server, we are getting this file from server and decrypting

**Reply by twiss** [reply]
Ah, sorry, I expected a zip file containing an encrypted messages :) In any case, indeed this file is not integrity protected. See https://github.com/openpgpjs/openpgpjs/discussions/1520.

**Reply by plabelclothing** [reply]
thank you! but this config doesn't work for me 
`{ allowInsecureDecryptionWithSigningKeys: true, allowUnauthenticatedMessages: true }`

**Reply by plabelclothing** [reply]
@twiss Hi =) Do you have any updates?

**Reply by larabr** [reply]
It's hard to debug without having the key. Just as sanity check, can you confirm the fingerprint of the encryption key is `DAD4 E948 C0B8 69D4` ? (See result of `(await privateKey.getEncryptionKey()).getKeyID().toHex()`).
 
If that is correct, could you run the following and report which error you get?
```js
const message = await openpgp.readMessage({ binaryMessage });
const encryptedSessionKey = message.packets[0];
const { keyPacket: encryptionKeyPacket } = await privateKey.getEncryptionKey();
await encryptedSessionKey.decrypt(encryptionKeyPacket);
```

**Reply by plabelclothing** [reply]
@larabr  hi, i confirm that the fingerprint of the encryption key is DAD4 E948 C0B8 69D4. I didn't get any errors

**Reply by twiss** [reply]
Hey :wave: Could you post the full code (updated with the config) that you're using to decrypt the message?

**Reply by plabelclothing** [reply]
@twiss Hey 👋  
`	const publicKeyEncode = fs.readFileSync('...').toString();
	const privateKeyEncode = fs.readFileSync('...').toString();
	const binaryMessage = fs.readFileSync('...');
	const publicKey = await openpgp.readKey({armoredKey: publicKeyEncode});

	const privateKey = await openpgp.readPrivateKey({armoredKey: privateKeyEncode});

	const message = await openpgp.readMessage({ binaryMessage });

	const decrypted = await openpgp.decrypt({
		decryptionKeys: privateKey, verificationKeys: publicKey,
		message,
		format: 'binary',
		config: { allowInsecureDecryptionWithSigningKeys: true, allowUnauthenticatedMessages: true }
	});
	console.log(decrypted.data); // Uint8Array`

**Reply by twiss** [reply]
Thanks. And could you post a full stack trace of the error you get?

**Reply by plabelclothing** [reply]
@twiss Hey, `Error: Error decrypting message: Session key decryption failed.
    at yc.decryptSessionKeys (/Users/aleks.ziabko/Documents/projects/PAYMENTO/Projects/backend-ing-payout/node_modules/openpgp/dist/node/openpgp.min.js:2:338227)
    at async yc.decrypt (/Users/aleks.ziabko/Documents/projects/PAYMENTO/Projects/backend-ing-payout/node_modules/openpgp/dist/node/openpgp.min.js:2:335327)
    at async Object.exports.decrypt (/Users/aleks.ziabko/Documents/projects/PAYMENTO/Projects/backend-ing-payout/node_modules/openpgp/dist/node/openpgp.min.js:16:132521)
    at async yy (/Users/aleks.ziabko/Documents/projects/PAYMENTO/Projects/backend-ing-payout/index.js:79:21)`

**Reply by twiss** [reply]
OK, thanks. I'm not entirely sure what's causing this error, but if the code that Lara sent earlier works, then you could try to use the result of that to decrypt the message, by passing it as `sessionKeys` to `openpgp.decrypt`?

**Reply by plabelclothing** [reply]
@twiss unfortunately, it doesn't  work Error: Decryption error
```			const publicKey = await openpgp.readKey({armoredKey: publicKeyEncode});
			const privateKey = await openpgp.readPrivateKey({armoredKey: privateKeyEncode});

			const message = await openpgp.readMessage({ binaryMessage: dataFile });
			const encryptedSessionKey = message.packets[0];
			const { keyPacket: encryptionKeyPacket } = await privateKey.getEncryptionKey();
			const sessionKeys = await encryptedSessionKey.decrypt(encryptionKeyPacket);

			let opt = {
				decryptionKeys: privateKey,
				verificationKeys: publicKey,
				sessionKeys,
				message,
				format: 'binary',
				config: {
					allowInsecureDecryptionWithSigningKeys: true,
					allowUnauthenticatedMessages: true,
				},
			};
			openpgp.decrypt(opt)
				.then((plaintext) => {
					fs.writeFileSync(config.directories.assets.private.pathZip + fileZip, plaintext.data);
					return resolve(true);
				})

**Reply by twiss** [reply]
Could you try something like:
```js
            const privateKey = await openpgp.readPrivateKey({armoredKey: privateKeyEncode});

            const message = await openpgp.readMessage({ binaryMessage: dataFile });
            const encryptedSessionKey = message.packets[0];
            const { keyPacket: encryptionKeyPacket } = await privateKey.getEncryptionKey();
            await encryptedSessionKey.decrypt(encryptionKeyPacket);

            let opt = {
                sessionKeys: [{
                    data: encryptedSessionKey.sessionKey,
                    algorithm: ({
                        1: 'idea',
                        2: 'tripledes',
                        3: 'cast5',
                        4: 'blowfish',
                        7: 'aes128',
                        8: 'aes192',
                        9: 'aes256',
                        10: 'twofish'
                    })[encryptedSessionKey.sessionKeyAlgorithm]
                }],
                verificationKeys: publicKey,
                message,
                format: 'binary',
                config: {
                    allowInsecureDecryptionWithSigningKeys: true,
                    allowUnauthenticatedMessages: true,
                },
            };
            openpgp.decrypt(opt)
                .then((plaintext) => {
                    fs.writeFileSync(config.directories.assets.private.pathZip + fileZip, plaintext.data);
                    return resolve(true);
                })
```
(it's a bit hacky, but just to check if it works). If it doesn't work, please post the stack trace.

**Reply by plabelclothing** [reply]
@twiss it doesn't work =(

**Reply by plabelclothing** [reply]
@twiss Hey! I checked my key, everything looks good 
`
# gpg -vvv sftp-ing-gpg.priv-nopass.asc 2>&1 | grep flags
        hashed subpkt 27 len 1 (key flags: 03)
        hashed subpkt 27 len 1 (key flags: 0C)
# gpg -vvv sftp-ing-gpg.pub.asc 2>&1 | grep flags
        hashed subpkt 27 len 1 (key flags: 03)
        hashed subpkt 27 len 1 (key flags: 0C)`

**Reply by larabr** [reply]
We need the stack trace of the error for the code posted by Daniel to debug further :)

**Reply by plabelclothing** [reply]
@larabr @twiss i'm sorry, it is my mistake,         const sessionKeys = await encryptedSessionKey.decrypt(encryptionKeyPacket); does't work =( 

`Error: Decryption error
    at /Users/mbazych/Downloads/test-openpgp/node_modules/openpgp/dist/node/openpgp.min.js:2:164760
    at async So.decrypt (/Users/mbazych/Downloads/test-openpgp/node_modules/openpgp/dist/node/openpgp.min.js:2:285034)
    at async yy (/Users/mbazych/Downloads/test-openpgp/index.js:57:29)`

**Reply by larabr** [reply]
Can you share what this returns: `console.log(privateKey.getKeyIDs().map(keyID => keyID.toHex()))`

**Reply by plabelclothing** [reply]
@larabr dad4e948c0b869d4,c1247152b3513612

**Reply by larabr** [reply]
Ok, then try this, and post the stack trace if it fails:
```js
const message = await openpgp.readMessage({ binaryMessage: dataFile });
const encryptedSessionKey = message.packets[0];
await encryptedSessionKey.decrypt(privateKey.keyPacket);
```

**Reply by plabelclothing** [reply]
@larabr thank you!!!! everything is working good

**Reply by larabr** [reply]
It's odd though that `openpgp.decrypt` failed, I think something might be off with the key certification signatures. Could you maybe share the public key (`privateKey.toPublic().armor()`)?
